docs/guides/nextauth-migration.mdx

```diff
@@ -0,0 +1,253 @@
+// With Clerk
+
+import { ClerkProvider } from "@clerk/nextjs";
+
+export default function App({ Component, pageProps }) {
+  return (
+    <ClerkProvider {...pageProps}>
+      <Component {...pageProps} />
+    </ClerkProvider>
+  );
+}
+```
+
+**Middleware**
+
+Replace NextAuth middleware with clerkMiddleware, use `protectedRoutes` as needed.
+
+```tsx
+// With NextAuth.js
+
+export { default } from "next-auth/middleware"
+
+export const config = { matcher: [/** private routes */] }
+```
+
+```tsx
+// With Clerk
+import { clerkMiddleware, createRouteMatcher } from "@clerk/nextjs/server";
+import { NextResponse } from "next/server";
+
+// opt into protection by adding routes to this array.
+const isProtectedRoute = createRouteMatcher([/** private routes */]);
+
+export default clerkMiddleware((auth, req) => {
+  // Check if the user visiting a protected route is logged in
+  // If not, redirect the user to the sign-in route
+  if (!auth().userId && isProtectedRoute(req)) {
+    return auth().redirectToSignIn();
+  }
+
+  // User isn't logged in and the route is public -- let them visit it
+  return NextResponse.next();
+});
+
+export const config = {
+  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+};
+```
+
+NextAuth's middleware always rejects unauthorized requests, and you might be using the Next.js middleware config to protect specific private routes. Clerk's middleware can run for your entire application and gives you fine-grained control over handling the authenticated state. In this snippet, `createRouteMatcher` is used to define routes that should be protected, and the callback function checks for protected routes and redirects the user to a sign in page.
+
+**Rendering**
+
+To conditionally render UI when the user is signed in, wrap it with `<SignedIn>`
+
+To conditionally render UI when the user is NOT signed in, wrap it with `<SignedOut>`
+
+```tsx
+// With Clerk
+
+<SignedIn>
+  <div>You are signed in</div>
+</SignedIn>
+<SignedOut>
+  <div>You are signed out</div>
+</SignedOut>
+```
+
+## Reading user and session data
+
+Replace any `getServerSession(req, res, authOptions)` to `getAuth(req)` from Clerk.
+
+```tsx
+// With NextAuth.js
+
+export async function getServerSideProps(context) {
+  const session = await getServerSession(context.req, context.res, authOptions)
+	...
+```
+
+```tsx
+// With Clerk
+
+export async function getServerSideProps(context) {
+  const session = getAuth(context.req);
+	...
+	return { props: { ...buildClerkProps(ctx.req) } };
+};
+```
+
+Replace NextAuth's `useSession` with Clerk's `useAuth` (or `useUser` if you need the full user object)
+
+```tsx
+// With NextAuth.js
+
+import { useSession } from "next-auth/react"
+...
+const { data: session } = useSession()
+```
+
+```tsx
+// With Clerk
+
+import { useAuth, useUser } from "@clerk/nextjs";
+...
+const { userId, sessionId, getToken } = useAuth();
+const { isSignedIn, user } = useUser();
+```
+
+**User IDs as Foreign Keys**       
```

Comment:
```suggestion
### User IDs as Foreign Keys    
```

Reply:
There should be a paragraph here that talks about Clerk's userId, and that there are different ways to resolve this when migration.

Reply:
The requested paragraph is missing.